packages/camera/camera_windows/windows/capture_controller.cpp

```diff
@@ -479,6 +479,13 @@ HRESULT CaptureControllerImpl::FindBaseMediaTypes() {
     return E_FAIL;
   }
 
+  hr = source->SetCurrentDeviceMediaType(
```

Comment:
It seems extremely surprising to have a method called `FindBaseMediaTypes()` have the side-effect of *setting* a media type. Why is this method the place where this is called?

Reply:
[`SetCurrentDeviceMediaType`](https://learn.microsoft.com/en-us/windows/win32/api/mfcaptureengine/nf-mfcaptureengine-imfcapturesource-setcurrentdevicemediatype) is a class function of `IMFCaptureSource` which was conveniently already accessible in the function scope `FindBaseMediaType`. For the bug fix, the function only needs to be called after `base_preview_media_type_` is set, which is when `FindBestMediaType` is called with it.

We can alternatively move it inside the function `CaptureControllerImpl::StartPreview` after `base_preview_media_type_` is set, but then we would need to prepare a `IMFCaptureSource` object for this scope.

Reply:
> For the bug fix, the function only needs to be called after `base_preview_media_type_` is set, which is when `FindBestMediaType` is called with it.

My primary concern is the maintainability of this code in the long term, rather than minimizing the diff. Having a method indicate that it gets a value have a side effect of changing a value is the kind of thing that makes maintaining code correctly a lot harder over the longer term.

> We can alternatively move it inside the function `CaptureControllerImpl::StartPreview`

That sounds like a much clearer location for setting values related to displaying the preview.